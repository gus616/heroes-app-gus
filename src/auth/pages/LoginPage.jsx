import { useContext } from "react";
import { useNavigate } from "react-router-dom"
import { AuthContext } from "../context";

export const LoginPage = () => {
    const navigate = useNavigate();
    const { login } = useContext(AuthContext);
    const onLogin = () => {

        const lastPath = localStorage.getItem('lastPath') || '/';

        login('Gus');

        navigate( lastPath, {
            replace: true
        });

    }
    return (
        <>
            <h1>Login Page</h1>
            <hr />

            <button className="btn btn-primary" onClick={onLogin}>Login</button>
        </>
    )
}