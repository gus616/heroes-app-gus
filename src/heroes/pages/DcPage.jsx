import { HeroList } from "../components/HeroList";

export const DcPage = () => {
    return (
        <>
            <h1 className="animate__animated animate__slideInDown mt-3 mb-4">DC Page</h1>
            <ul className="animate__animated animate__pulse">
                <HeroList publisher="DC Comics"/>
            </ul>
            
        </>
    )
};