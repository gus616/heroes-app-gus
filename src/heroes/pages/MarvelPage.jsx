import { HeroList } from '../components/HeroList';

export const MarvelPage = () => {

    return (
        <>
            <h1 className="animate__animated animate__slideInDown mt-3 mb-4">Marvel Page</h1>
           <ul className="animate__animated animate__pulse">
              <HeroList publisher='Marvel Comics' />
           </ul>
      
         
        </>
    )
};