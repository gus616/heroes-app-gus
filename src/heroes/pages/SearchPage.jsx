import { useLocation, useNavigate } from 'react-router-dom';
import { useForm } from '../../hooks/useForm';
import { HeroCard } from '../components/HeroCard';
import queryString from 'query-string';
import { getHeroByName } from '../helpers/';
export const SearchPage = () => {

    const { searchText, onInputChange, onResetForm } = useForm({searchText: '', });

    const navigate = useNavigate();
    const location = useLocation();

    const {q = ''} = queryString.parse(location.search);

    const heroes = getHeroByName(q);

    const onSearchHandler = (e) => {
        e.preventDefault(); 

       if( searchText.trim().length <= 1 ) return;

       navigate(`?q=${searchText.toLowerCase().trim()}`);   
       
       onResetForm();
    };    

    return(
        <>
            <h1> Search </h1>
            <hr/>
            <div className="row">
                <div className="col-5">
                    <h4>Searching</h4>
                    <hr/>
                    <form onSubmit={onSearchHandler} aria-label="form">
                        <input
                            type="text"
                            placeholder="search a hero"
                            className="form-control"
                            name="searchText"
                            value = { searchText }
                            onChange = { onInputChange }
                            autoComplete="off"/>

                        <button 
                            className="btn btn-dark mt-3 mb-3"
                            > Search
                        </button>
                    </form>
                </div>
                <div className="col-7">
                    <h4>Results</h4>
                    <hr/>

                    {
                        (q === '' && 
                            <div className="alert alert-primary">
                                Search a Hero
                            </div>) 
                    }

                    {
                        (q !== ''  &&  heroes.length <= 0  && 
                            <div className="alert alert-danger" aria-label="no-results">
                                There is no results with <b>{q}</b>.
                            </div>)
                    }                    
       
                   {
                        heroes.map((hero) => (<HeroCard key={hero.id} {...hero} />))
                   }
                </div>
            </div>
        </>
    )
};