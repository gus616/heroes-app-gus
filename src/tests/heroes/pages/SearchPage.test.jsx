import {cleanup, fireEvent, render, screen} from '@testing-library/react';
import { MemoryRouter, useNavigate } from 'react-router-dom';
import { SearchPage } from '../../../heroes/pages/SearchPage';
const mockedUseNavigate = jest.fn();

 
describe('<SearchPage /> tests', () => {
    beforeEach(() => jest.clearAllMocks());
   
    
    test('must show with default values', () => {

       const { container } = render(<MemoryRouter>          
                <SearchPage />        
        </MemoryRouter>);

        expect( container ).toMatchSnapshot();
    });


    test('must show Batman after input and clicking search button', () => {
        render(
            <MemoryRouter initialEntries={['/search']}>
                <SearchPage />
            </MemoryRouter>
        );

        const input = screen.getByRole('textbox');
        fireEvent.change(input, {target: {value: 'batman'}});

        const btn = screen.getByRole('button');
        fireEvent.click(btn);

        const img = screen.getByRole('img');
        

        expect( img.src ).toBe('http://localhost/assets/heroes/dc-batman.jpg');

    });

    test('must show an error if hero not found', () => {
        render(
            <MemoryRouter initialEntries={['/search']}>
                <SearchPage />
            </MemoryRouter>
        );

        const input = screen.getByRole('textbox');
        fireEvent.change(input, {target: {value: 'sdfnksdfnk'}});

        const btn = screen.getByRole('button');
        fireEvent.click(btn);
        //screen.debug();
    
       expect( screen.getByLabelText("no-results").innerHTML.includes("There is no results with")).toBeTruthy();

    });

/* 
    test('must call navigate to new page with superman value using input and calling submit', () => {   

        //arrange

        
       
           
        render(
            <MemoryRouter initialEntries={['/search']}>
                <SearchPage />
            </MemoryRouter>
        );

        //act
        const input = screen.getByRole('textbox');
        fireEvent.change(input, {target: {value: 'superman'}});



        const form = screen.getByRole('form');
        fireEvent.submit(form);

        screen.debug();

   

        //assert
        expect(mockedUseNavigate).toHaveBeenCalled();
        //expect(mockedUseNavigate).toHaveBeenCalledWith('/search?q=superman', {"replace": true});
     
    }); */


   
});