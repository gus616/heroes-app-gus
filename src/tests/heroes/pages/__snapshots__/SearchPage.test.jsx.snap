// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchPage /> tests must show with default values 1`] = `
<div>
  <h1>
     Search 
  </h1>
  <hr />
  <div
    class="row"
  >
    <div
      class="col-5"
    >
      <h4>
        Searching
      </h4>
      <hr />
      <form
        aria-label="form"
      >
        <input
          autocomplete="off"
          class="form-control"
          name="searchText"
          placeholder="search a hero"
          type="text"
          value=""
        />
        <button
          class="btn btn-dark mt-3 mb-3"
        >
           Search
        </button>
      </form>
    </div>
    <div
      class="col-7"
    >
      <h4>
        Results
      </h4>
      <hr />
      <div
        class="alert alert-primary"
      >
        Search a Hero
      </div>
    </div>
  </div>
</div>
`;
